Tests/IntegrationTests/BasicTests.swift

```diff
@@ -27,13 +27,25 @@ private struct BasicTests {
             Tag.Feature.Command.Build,
         ),
     )
-    func testExamplePackageDealer() throws {
-        try withTemporaryDirectory { tempDir in
+    func testExamplePackageDealer() async throws {
+        try await withTemporaryDirectory { tempDir in
             let packagePath = tempDir.appending(component: "dealer")
             withKnownIssue(isIntermittent: true) {
-                // marking as withKnownIssue(intermittent: trye) as git operation can fail.
-                try sh("git\(ProcessInfo.exeSuffix)", "clone", "https://github.com/apple/example-package-dealer", packagePath)
+                // marking as withKnownIssue(intermittent: true) as git operation can fail.
+
+                #if os(macOS)
+                    // On macOS, we add the HOME variable to avoid git errors.                
+                    try sh("git\(ProcessInfo.exeSuffix)", "clone", "https://github.com/apple/example-package-dealer", packagePath, env: ["HOME": tempDir.pathString])
+                #else
+                    try sh("git\(ProcessInfo.exeSuffix)", "clone", "https://github.com/apple/example-package-dealer", packagePath)
+                #endif
+            }
+
+            // Do not run the test when the git clone operation failed
+            if !FileManager.default.fileExists(atPath: packagePath.pathString) {
+                return
```

Comment:
I'm not fond of this as it gives the impression the tests passes, when it wasn't executed at all.  We should include a TODO to use the *Test cancellation* feature once it's available.

https://forums.swift.org/t/pitch-test-cancellation/81847/18

In the meantime, can we record a warning `Issue.record("....", severity: .warning)` as introduced by https://forums.swift.org/t/accepted-st-0013-test-issue-severity/81385/2 (though our self-hosted pipelines doesn't currently use a version of the toolchain that has these features)

Reply:
In what version of Swift the `Issue.record("....", severity: .warning)` is introduced ? Can I safely introduce that change, knowing it will not cause additional issues on Swift.org CI ?